jms/org.wso2.carbon.transport.jms/src/main/java/org/wso2/carbon/transport/jms/factory/JMSClientConnectionFactory.java

```diff
@@ -0,0 +1,226 @@
+ * under the License.
+ */
+
+package org.wso2.carbon.transport.jms.factory;
+
+import org.apache.commons.pool2.impl.GenericObjectPool;
+import org.apache.commons.pool2.impl.GenericObjectPoolConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.carbon.transport.jms.exception.JMSConnectorException;
+import org.wso2.carbon.transport.jms.sender.sessionpool.SessionPoolFactory;
+import org.wso2.carbon.transport.jms.sender.wrappers.ConnectionWrapper;
+import org.wso2.carbon.transport.jms.sender.wrappers.SessionWrapper;
+import org.wso2.carbon.transport.jms.utils.JMSConstants;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+import javax.jms.Connection;
+import javax.jms.ExceptionListener;
+import javax.jms.JMSException;
+import javax.jms.XAConnection;
+
+/**
+ * Extended class to handle Client side Connection Factory requirements
+ */
+public class JMSClientConnectionFactory extends JMSConnectionResourceFactory {
+
+    private static final Logger logger = LoggerFactory.getLogger(JMSClientConnectionFactory.class);
+    /**
+     * Default Wait timeout (in milliseconds) for the Session pool
+     */
+    private static final int poolWaitTimeout = 30 * 1000;
+    /**
+     * Default Size of the Connection list
+     */
+    private int maxNumberOfConnections = 5;
+    /**
+     * Default Number of session per Connection
+     */
+    private int maxSessionsPerConnection = 10;
+    /**
+     * Indicates cache level given by the user.
+     */
+    private boolean clientCaching = true;
+
+    /**
+     * List of Connection wrapper objects
+     */
+    private List<ConnectionWrapper> connections = null;
+
+    /**
+     * Pool of Session wrapper objects
+     */
+    private GenericObjectPool<SessionWrapper> sessionPool = null;
+
+    /**
+     * Constructor
+     *
+     * @param properties JMS properties
+     * @throws JMSConnectorException
+     */
+    public JMSClientConnectionFactory(Properties properties) throws JMSConnectorException {
+        super(properties);
+        setCache(properties);
+
+        // session pool configurations
+        if (properties.getProperty(JMSConstants.PARAM_MAX_CONNECTIONS) != null) {
+            try {
+                this.maxNumberOfConnections = Integer
+                        .parseInt(properties.getProperty(JMSConstants.PARAM_MAX_CONNECTIONS));
+            } catch (NumberFormatException ex) {
+                logger.error("Non-integer value configured for JMS Client Connection count");
+            }
+        }
+        if (properties.getProperty(JMSConstants.PARAM_MAX_SESSIONS_ON_CONNECTION) != null) {
+            try {
+                this.maxSessionsPerConnection = Integer
+                        .parseInt(properties.getProperty(JMSConstants.PARAM_MAX_SESSIONS_ON_CONNECTION));
+            } catch (NumberFormatException ex) {
+                logger.error("Non-integer value configured for JMS Client Sessions count");
```

Comment:
Shall we log the exception trace as well?

Reply:
added